blueocean-plugin/src/main/resources/archetype-resources/src/main/java/Computers.java

```diff
@@ -0,0 +1,163 @@
+        Jenkins j = Jenkins.getInstance();
+        for (Computer c : j.getComputers()) {
+            info.add(new ComputerInfo(getLink(), c));
+        }
+        return info.toArray(new ComputerInfo[info.size()]);
+    }
+
+    @ExportedBean
+    public static class ComputerInfo extends Resource {
+        final Link parent;
+        Computer computer;
+
+        public ComputerInfo(Link parent, Computer computer) {
+            this.parent = parent;
+            this.computer = computer;
+        }
+
+        @Exported
+        public String getDisplayName() {
+            return computer.getDisplayName();
+        }
+
+        @Exported(inline=true)
+        public Container<ExecutorInfo> getExecutors() {
+            final List<ExecutorInfo> out = new ArrayList<>();
+            if (computer != null) {
+                for (Executor e : computer.getExecutors()) {
+                    out.add(new ExecutorInfo(this, e));
+                }
+            }
+            return new Container() {
+                @Override
+                public Object get(String string) {
+                    throw new UnsupportedOperationException("Not supported.");
+                }
+
+                @Override
+                public Iterator iterator() {
+                    return out.iterator();
+                }
+
+                @Override
+                public Link getLink() {
+                    return parent.rel("executors");
+                }
+            };
+        }
+
+        @Override
+        public Link getLink() {
+            return parent.rel("computer");
+        }
+    }
+    
+    @ExportedBean
+    public static class ExecutorInfo extends Resource {
+        final Resource parent;
+        final Executor executor;
+        private ExecutorInfo(Resource parent, Executor executor) {
+            this.parent = parent;
+            this.executor = executor;
+        }
+
+        @Override
+        public Link getLink() {
+            return parent.getLink().rel("executor");
+        }
+        
+        @Exported
+        public String getDisplayName() {
+            return executor.getDisplayName();
+        }
+        
+        @Exported
+        public boolean isIdle() {
+            return executor.isIdle();
+        }
+
+        @Exported(inline=true)
+        public BlueRun getRun() {
+            Queue.Executable e = executor.getCurrentExecutable();
+            if (e == null) {
+                return null;
+            }
+            Run r = null;
+            SubTask subTask = e.getParent();
+            if (subTask instanceof ExecutorStepExecution.PlaceholderTask) {
+                ExecutorStepExecution.PlaceholderTask task = (ExecutorStepExecution.PlaceholderTask)subTask;
+                r = task.run();
+            }
+            if (e instanceof WorkflowRun) {
+                r = (WorkflowRun)e;
+            }
+            if (e instanceof AbstractBuild) {
+                r = (AbstractBuild)e;
```

Comment:
Uh, why not just

```java
if (e instance of Run) {
    r = (Run) e;
}
```

? And then you would not need a compile-time dependency on `workflow-job`. Which you should not have.